File: java/src/test/java/com/kcl/PingTest.java
package com.kcl;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.PingArgs;
import com.kcl.api.Spec.PingResult;

public class PingTest {
    @Test
    public void testPing() throws Exception {
        API api = new API();
        PingArgs args = PingArgs.newBuilder().setValue("hello-kcl").build();
        PingResult result = api.ping(args);
        Assert.assertEquals("hello-kcl", result.getValue());
    }
}

File: kotlin/src/test/kotlin/com/kcl/PingTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.pingArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class PingTest {
    @Test
    fun testPing() {
        val args = pingArgs {
            value = "hello-kcl"
        }
        val api = API()
        val result = api.ping(args)
        assertEquals("hello-kcl", result.value)
    }
}
